libs/store/src/lib/infrastructure/persistence/store/entity-store-server/http-url-generator.ts

```diff
@@ -0,0 +1,84 @@
+import { Injectable } from '@angular/core';
+import { Pluralizer } from '@ngrx/data';
+
+import { EntityOperation, EntityPath } from '../../../../domain/rest-resource';
+
+/**
+ * Generate the base part of an HTTP URL for
+ * single entity or entity collection resource
+ */
+export abstract class HttpUrlGenerator {
+    /**
+     * Return the base URL for a single entity resource,
+     * e.g., the base URL to get a single hero by its id
+     */
+    abstract entityResource(entityName: string, root: string, operation: EntityOperation, path?: EntityPath): string;
+
+    /**
+     * Return the base URL for a collection resource,
+     * e.g., the base URL to get all heroes
+     */
+    abstract collectionResource(
+        entityName: string,
+        root: string,
+        operation: EntityOperation,
+        path?: EntityPath
+    ): string;
+}
+
+/**
+ * HTTP URl Generator
+ *
+ * This implementation takes into account the Entity meta options
+ *
+ */
+@Injectable()
+export class DefaultHttpUrlGenerator implements HttpUrlGenerator {
+    constructor(private pluralizer: Pluralizer) {}
+
+    entityResource(entityName: string, root: string, operation: EntityOperation, path?: EntityPath): string {
+        return this.getEntityResourceUrl(entityName, root, operation, path);
+    }
+
+    collectionResource(entityName: string, root: string, operation: EntityOperation, path?: EntityPath): string {
+        return this.getEntityResourceUrl(entityName, root, operation, path, true);
+    }
+
+    protected getEntityResourceUrl(
+        entityName: string,
+        root: string,
+        operation: EntityOperation,
+        entityPath: EntityPath | undefined,
+        pluralize = false
+    ): string {
+        let path: string = pluralize ? this.pluralizer.pluralize(entityName) : entityName;
+
+        if (entityPath) {
+            path = this.getEntityOperationPath(entityPath, operation) || path;
+        }
+
+        return `${normalize(root)}/${normalize(path)}/`.toLowerCase();
+    }
```

Comment:
Are we going to support subresource paths, where there is a parent id in the URL like `GET /requisitions/123/lineitems`? If so, won't we need to pass in the entity itself as input?

Reply:
Yes, you are right, we need it.
Can we add this functionality a little bit later?

Reply:
I think this is one of the core functionality we need to support. but I aggree to get moving, we can add this as next task. not really later but more next PR